modules/core/02-client/migrations/v7/genesis.go

```diff
@@ -0,0 +1,82 @@
+package v7
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	"github.com/cosmos/ibc-go/v6/modules/core/exported"
+)
+
+// MigrateGenesis accepts an exported IBC client genesis file and migrates it to:
+//
+// - Update solo machine client state protobuf definition (v2 to v3)
+// - Remove all solo machine consensus states
+// - Remove localhost client
+func MigrateGenesis(clientGenState *types.GenesisState, cdc codec.ProtoCodecMarshaler) (*types.GenesisState, error) {
+	// To prune the client and consensus states, we will create new slices to fill up
+	// with information we want to keep.
+	var (
+		clientsConsensus []types.ClientConsensusStates
+		clients          []types.IdentifiedClientState
+	)
+
+	for _, client := range clientGenState.Clients {
+		clientType, _, err := types.ParseClientIdentifier(client.ClientId)
+		if err != nil {
+			return nil, err
+		}
+
+		// update solo machine client state defintions
+		switch clientType {
+		case exported.Solomachine:
+			var clientState ClientState
+			if err := cdc.Unmarshal(client.ClientState.Value, &clientState); err != nil {
+				return nil, sdkerrors.Wrap(err, "failed to unmarshal client state bytes into solo machine client state")
+			}
+
+			updatedClientState := migrateSolomachine(clientState)
+
+			any, err := types.PackClientState(&updatedClientState)
```

Comment:
`any` is now a golang keyword (alias for `interface{}` as of golang 1.18) can we change this to maybe `anyCodec` or similar?

Reply:
nice catch, we could use `protoAny` to avoid shadowing maybe? I reckon `any` is used widely across ibc-go and the sdk as well. 

I think I noticed before we also. use `cap` for capability in places which is the "capacity" function in Go. I don't think its a super big deal tho and shouldn't cause any issues

Reply:
updated